source/extensions/clusters/aggregate/cluster.cc

```diff
@@ -0,0 +1,72 @@
+#include "extensions/clusters/aggregate/cluster.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+Cluster::Cluster(const envoy::api::v2::Cluster& cluster,
+                 const envoy::config::cluster::aggregate::ClusterConfig& config,
+                 Upstream::ClusterManager& cluster_manager, Runtime::Loader& runtime,
+                 Server::Configuration::TransportSocketFactoryContext& factory_context,
+                 Stats::ScopePtr&& stats_scope, bool added_via_api)
+    : Upstream::BaseDynamicClusterImpl(cluster, runtime, factory_context, std::move(stats_scope),
+                                       added_via_api),
+      cluster_manager_(cluster_manager) {
+  for (const auto& cluster : config.clusters()) {
+    clusters_.emplace_back(cluster);
+  }
+}
+
+Upstream::ThreadLocalCluster* Cluster::getThreadLocalCluster(const std::string& name) const {
+  Upstream::ThreadLocalCluster* tlc = cluster_manager_.get(name);
+  if (tlc == nullptr) {
+    throw EnvoyException(fmt::format("no thread local cluster with name {}", name));
+  }
+
+  return tlc;
+}
+
+std::pair<Upstream::PrioritySetImpl, std::vector<Upstream::ThreadLocalCluster*>>
+Cluster::linearizePrioritySet() const {
+  Upstream::PrioritySetImpl priority_set;
+  std::vector<Upstream::ThreadLocalCluster*> priority_to_cluster;
+  int next_priority = 0;
+
+  for (const auto& cluster : clusters_) {
+    auto tlc = getThreadLocalCluster(cluster);
+    for (const auto& host_set : tlc->prioritySet().hostSetsPerPriority()) {
+      if (host_set->hosts().size() > 0) {
+        priority_set.updateHosts(
```

Comment:
you'll need to update this value whenever the TLS priority set changes, you'll want to hook into `addPriorityUpdateCb` on the TLS HostSet

Reply:
Callback added.